### PR TITLE
[merge jam] Master port 49261 - consul modules

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Master port #49261

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Review by waynew — CHANGES_REQUESTED** [review]
This does need a `changelog` file. Otherwise 👍

**Review by krionbsd — DISMISSED** [review]

**Review by twangboy — DISMISSED** [review]

**Review by waynew — DISMISSED** [review]

**Review by waynew — DISMISSED** [review]
I'm pretty sure those were flaky test failures. Re-running the tests.

**Comment by waynew** [comment]
@garethgreenaway looks like black/pre-commit failures, wanna run that locally?

File: salt/modules/consul.py [hunk 1]
```diff
@@ -2155,7 +2170,7 @@ def acl_update(consul_url=None, token=None, **kwargs):
         ret["message"] = "ACL {} created.".format(kwargs["name"])
     else:
         ret["res"] = False
-        ret["message"] = "Adding ACL {} failed.".format(kwargs["name"])
+        ret["message"] = "Updating ACL " "{} failed.".format(kwargs["name"])
```
Comment (Ch3LL) [inline salt/modules/consul.py:2173]:
```suggestion
        ret["message"] = "Updating ACL {} failed.".format(kwargs["name"])
```

File: salt/states/consul.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
+"""
+Consul Management
+=================
+
```
Comment (Ch3LL) [inline salt/states/consul.py:4]: Can you add a versioned added for 3004?

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: salt/states/consul.py (hunk 2 above)
Comment (garethgreenaway) [reply salt/states/consul.py:4]: Should be 3005 right?

Comment (Ch3LL) [reply salt/states/consul.py:4]: yeah lol

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**